C++ API parity: AdaptiveAvgPool1d

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26915 C++ API parity: MaxUnpool2d
* #26896 C++ API parity: MaxUnpool1d
* #26819 C++ API parity: AdaptiveAvgPool3d
* #26818 C++ API parity: AdaptiveAvgPool2d
* **#26808 C++ API parity: AdaptiveAvgPool1d**

Differential Revision: [D17627827](https://our.internmc.facebook.com/intern/diff/D17627827)

pbelevich committed:
C++ API parity: AdaptiveAvgPool1d

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"




[ghstack-poisoned]

yf225 reviewed (approved):
Looks awesome, thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"


Differential Revision: [D17627827](https://our.internmc.facebook.com/intern/diff/D17627827)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"


Differential Revision: [D17627827](https://our.internmc.facebook.com/intern/diff/D17627827)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"


Differential Revision: [D17627827](https://our.internmc.facebook.com/intern/diff/D17627827)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveAvgPool1d"


Differential Revision: [D17627827](https://our.internmc.facebook.com/intern/diff/D17627827)

[ghstack-poisoned]
